### PR TITLE
Update dependencies for homie-device

### PR DESCRIPTION
**Pull request by dwrobel**
Signed-off-by: Damian Wrobel <dwrobel@ertelnet.rybnik.pl>

### CODE REVIEW

File: homie-device/Cargo.toml [hunk 1]
```diff
@@ -11,17 +11,15 @@ categories = ["network-programming"]
...
+futures = "0.3.21"
 local_ipaddress = "0.1.3"
-log = "0.4.14"
-mac_address = "1.1.2"
-rumqttc = "0.10.0"
-tokio = "1.12.0"
-thiserror = "1.0.29"
+log = "0.4.16"
+mac_address = "1.1.3"
+rumqttc = "0.11.0"
```
Comment (alsuren) [inline homie-device/Cargo.toml:18]: Thanks for doing this.

Andrew points out that you also need to bump all other crates to this version of rumqttc, to fix the build.

I would expect our Cargo.lock to change as well. Make sure you check this in, to make the bin crate builds reproducible.

**Comment by dwrobel** [comment]
>Andrew points out that you also need to bump all other crates to this version of rumqttc, to fix the build.

The problem is that I'm using only homie-device crate and while I can successfully upgrade its dependencies to the latest available without any issue that's not the case for other crates in this repository.

**Comment by qwandor** [comment]
Well I'm afraid we can't take a patch that breaks the build. What's the matter when you try to upgrade the other crates?

**Comment by dwrobel** [comment]
>Well I'm afraid we can't take a patch that breaks the build.

I'm not trying to break your build in any way.

>What's the matter when you try to upgrade the other crates?

The following were successfully upgraded:
- homie-controller
- homie-device
- mijia

but those failed:
- homie-influx
- mijia

due to duplicated crates with different versions. See the example for homie-influx:

```
$ cargo tree -d
rustls v0.19.1
├── hyper-rustls v0.22.1
│   └── reqwest v0.11.3
│       └── influx_db_client v0.5.2
│           └── homie-influx v0.2.6 (/home/dw/projects/modbus/mijia-homie/homie-influx)
├── reqwest v0.11.3 (*)
└── tokio-rustls v0.22.0
    ├── hyper-rustls v0.22.1 (*)
    └── reqwest v0.11.3 (*)

rustls v0.20.6
├── homie-influx v0.2.6 (/home/dw/projects/modbus/mijia-homie/homie-influx)
└── tokio-rustls v0.23.4
    └── rumqttc v0.12.0
        ├── homie-controller v0.5.1 (/home/dw/projects/modbus/mijia-homie/homie-controller)
        │   └── homie-influx v0.2.6 (/home/dw/projects/modbus/mijia-homie/homie-influx)
        └── homie-influx v0.2.6 (/home/dw/projects/modbus/mijia-homie/homie-influx)

rustls-pemfile v0.3.0
└── rumqttc v0.12.0 (*)

rustls-pemfile v1.0.0
└── rustls-native-certs v0.6.2
    └── homie-influx v0.2.6 (/home/dw/projects/modbus/mijia-homie/homie-influx)

sct v0.6.1
└── rustls v0.19.1 (*)

sct v0.7.0
└── rustls v0.20.6 (*)

tokio-rustls v0.22.0 (*)

tokio-rustls v0.23.4 (*)

webpki v0.21.4
├── hyper-rustls v0.22.1 (*)
├── rustls v0.19.1 (*)
├── tokio-rustls v0.22.0 (*)
└── webpki-roots v0.21.1
    └── reqwest v0.11.3 (*)

webpki v0.22.0
├── rustls v0.20.6 (*)
└── tokio-rustls v0.23.4 (*)
```

**Comment by qwandor** [comment]
I've fixed the build failures from the `rustls` update. Having multiple versions of some dependencies in the tree is annoying but not a blocking issue as long as it still builds.

**Review by qwandor — APPROVED** [review]

**Comment by qwandor** [comment]
I'll make a new release soon, once I get a few other minor things in.

**Pull request merged**